src/main/java/com/onaio/steps/helper/DatabaseHelper.java

```diff
@@ -129,4 +126,10 @@ public boolean isColumnExist(SQLiteDatabase db, String tableName, String columnN
 
         return isExist;
     }
+
+    public void dropTable(SQLiteDatabase db, String... tableNames) {
+        for (String table : tableNames) {
+            db.execSQL(String.format("DROP TABLE IF EXISTS %s", table));
```

Comment:
This does not make it safe from SQL injection.

1. Evaluate the uses of `rawQuery` and `execSQL` in the repository and use parametirized queries eg, `execSQL(query, args)`, `bindString` and/or `compileStatement`. Figure out which ones are affected by user-input
2. Make the appropriate changes

Here is a Google article that explains more on preventing SQL injection https://sites.google.com/site/smsdproject/home/sql-injections/hands-on-lab-practice

Reply:
tried this `execSQL("DROP TABLE IF EXISTS  %s", Household.TABLE_NAME);` but getting exception. Additionally, we don't have security implications on that query because the query is static not dynamic.

Reply:
@owais-vd Yes, that should not work. You should have checked the link provided which provides more details. I still think that you should tackled #1 else I'll have to tackle it since it needs to go in the final report

Reply:
@owais-vd Sorry, I meant tackle this

> Evaluate the uses of `rawQuery` and `execSQL` in the repository and use parametirized queries eg, `execSQL(query, args)`, `bindString` and/or `compileStatement`. Figure out which ones are affected by user-input

Reply:
I tried `rawQuery` and `exceSQL` but both methods return `SQLiteException` with `bindArgs` and only work on data selection queries. I've also checked the suggestion link and they are also concatenating the `String` value in `execSQL` method instead `bindArgs`.

Reply:
Cool, have you tackled the first bullet here https://github.com/onaio/steps-app/pull/208#discussion_r875673134

Reply:
yeah, i also tried to evaluate the query with `compileStatement` with string binding but also got the same exception.

Reply:
Ok, it seems there's a misunderstanding. Let me rephrase

1. Search for uses of `rawQuery` and `execSQL` in the repository
2. Figure out which queries are affected by user-input
3. Convert them to use parametirized queries as given in the examples. If you don't convert them, share justification details on them so that we can use this in the report to describe why we still have such queries and the security implication

@owais-vd Does this make more sense?

Reply:
cool, that make sense. in the repository we are using both methods `rawQuery` and `execSQL` for static query i.e not required user-input, and none of the methods are affected by user-input i.e no user-input required in any query which is handled by `rawQuery` and `execSQL`. User input is required only for data insertion and update but we are using `db.insert()` and `db.update()` instead of a raw query.

Reply:
Cool, you can add this to the issue and provide evidence if need be. I think that's the main issue and we can get further recommendations/direction on these

Reply:
https://github.com/onaio/steps-app/issues/199#issuecomment-1131642537